src/rules.c

```diff
@@ -44,8 +44,15 @@ void rule_apply(struct rule *r, struct notification *n)
         }
         if (r->format)
                 n->format = r->format;
-        if (r->script)
-                n->script = r->script;
+        if (r->script){
+                n->scripts = g_renew(const char*,n->scripts,n->script_count + 1);
+                n->scripts[n->script_count] = r->script;
+
+                n->script_run = g_renew(bool,n->script_run,n->script_count + 1);
+                n->script_run[n->script_count] = false;
```

Comment:
There's no need to have this as an array, scripts are all run at the same time either way, so if one has ran, they all have.